into/utils.py

```diff
@@ -206,3 +211,82 @@ def ignoring(*exceptions):
 
 from multipledispatch import Dispatcher
 sample = Dispatcher('sample')
+
+
+@curry
+def pmap(f, iterable):
+    """Map `f` over `iterable` in parallel using a ``ThreadPool``.
+    """
+    p = ThreadPool()
+    try:
+        result = p.map(f, iterable)
+    finally:
+        p.terminate()
+    return result
+
+
+@curry
+def write(triple, writer):
+    """Write a file using the input from `gentemp` using `writer` and return
+    its index and filename.
+
+    Parameters
+    ----------
+    triple : tuple of int, str, str
+        The first element is the index in the set of chunks of a file, the
+        second element is the path to write to, the third element is the data
+        to write.
+
+    Returns
+    -------
+    i, filename : int, str
+        File's index and filename. This is used to return the index and
+        filename after splitting files.
+
+    Notes
+    -----
+    This could be adapted to write to an already open handle, which would
+    allow, e.g., multipart gzip uploads. Currently we open write a new file
+    every time.
+    """
+    i, filename, data = triple
+    with writer(filename, mode='wb') as f:
+        f.write(data)
+    return i, filename
+
+
+def gentemp(it, suffix=None, start=0):
+    """Generate an index, a temporary file, and return data for each element
+    in `it
+
+    Parameters
+    ----------
+    it : iterable
+    suffix : str
+        Suffix to add to each temporary file's name
+    """
+    for i, data in enumerate(it, start=start):  # aws needs parts to start at 1
+        _, fn = tempfile.mkstemp(suffix=suffix)
```

Comment:
`_` may be an open file handle.  Windows may not like it if other operations try to `open(fn)`